File: app2/src/main/java/guoan/app2/MyApplication.java
package guoan.app2;

import android.app.Activity;
import android.app.Application;
import android.content.Context;
import android.support.annotation.Nullable;
import android.widget.Toast;

import com.guoan.community.sdk.CommunityFactory;
import com.guoan.community.sdk.business.CommunityCallBack;
import com.guoan.community.sdk.business.CommunityJavaScriptInterface;
import com.guoan.community.sdk.business.LocationInfo;
import com.guoan.community.sdk.business.ShareInfo;
import com.guoan.community.sdk.business.StoreInfo;
import com.guoan.community.sdk.business.UserInfo;
import com.guoan.community.sdk.business.pay.CommunityPayment;

import java.util.List;

/**
 * @author andylove
 * @date 2018/3/1
 */

public class MyApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();
        CommunityFactory.Companion.getInstance().initSdkAuth(getApplicationContext());
        CommunityFactory.Companion.getInstance().initCallBack(new CommunityCallBack() {

            @Override
            public void onDoBeforFinished(@Nullable Activity context) {

            }

            @Override
            public void onShare(@Nullable Context context, @Nullable CommunityJavaScriptInterface jsInterface, @Nullable String responseId, @Nullable ShareInfo shareInfo) {

            }

            @Nullable
            @Override
            public List<StoreInfo> onGetStoreInfoList() {
                return null;
            }

            @Override
            public void onPay(@Nullable Context context, @Nullable CommunityJavaScriptInterface jsInterface, @Nullable String responseId, @Nullable CommunityPayment payment) {
                Toast.makeText(context, "去支付", Toast.LENGTH_SHORT).show();
            }

            @Override
            public void onJumpNative(Context context, String type, String param) {
                Toast.makeText(getApplicationContext(), "跳到国安社区的具体页面" + type + "----" + param, Toast.LENGTH_SHORT).show();
            }

            @Nullable
            @Override
            public StoreInfo onGetStoreInfo() {
                Toast.makeText(getApplicationContext(), "获取门店信息", Toast.LENGTH_SHORT).show();
                return null;
            }

            @Override
            public void onGetUserAddressList(Context context, CommunityJavaScriptInterface jsInterface, String responseId) {
                Toast.makeText(context, "异步获取地址列表", Toast.LENGTH_SHORT).show();
                jsInterface.callBackAddList(responseId, null);
            }

            @Override
            public LocationInfo onGetTempAddress() {
                Toast.makeText(getApplicationContext(), "获取位置信息", Toast.LENGTH_SHORT).show();
                return null;
            }

            @Override
            public UserInfo onGetUserInfo() {
                Toast.makeText(getApplicationContext(), "获取用户信息", Toast.LENGTH_SHORT).show();

                return null;
            }

            @Override
            public void onTryLogin(Context context, int reqCode) {
                Toast.makeText(context, "去登录", Toast.LENGTH_SHORT).show();
            }
        });
    }
}

File: app/src/main/java/com/guoan/sdkdemo/MyApplication.kt
package com.guoan.sdkdemo

import android.app.Activity
import android.app.Application
import android.content.Context
import android.content.Intent
import com.guoan.community.sdk.CommunityFactory
import com.guoan.community.sdk.business.*
import com.guoan.community.sdk.business.pay.CommunityPayment
import org.jetbrains.anko.toast

/**
 * Created by andylove on 2018/2/8.
 * demo application
 */
class MyApplication : Application() {

    companion object {
        var phone: String? = null
        var token: String? = null

        var lat: String? = null
        var lon: String? = null
        var cityCode: String? = null
        var sJsInterface: CommunityJavaScriptInterface? = null
        var sResponseId: String? = null
    }

    override fun onCreate() {
        super.onCreate()
        //集成国安社区
        CommunityFactory.getInstance()?.initSdkAuth(applicationContext)
        CommunityFactory.getInstance()?.initCallBack(object : CommunityCallBack {
            override fun onJumpNative(context: Context?, type: String?, param: String?) {
                toast("跳到国安社区的具体页面" + type + "----" + param)
            }

            override fun onGetStoreInfo(): StoreInfo? {
                toast("获取门店信息")
                return null
            }

            override fun onGetUserAddressList(context: Context?, jsInterface: CommunityJavaScriptInterface?, responseId: String?) {
                toast("异步获取地址列表信息")
                //获取用户地址列表后回调
                //jsInterface?.callBackAddList(responseId, null)
            }

            override fun onGetTempAddress(): LocationInfo? {
                toast("获取位置信息")
                var locationInfo = LocationInfo(lat, lon, cityCode)
                return locationInfo
            }

            override fun onTryLogin(context: Context?, reqCode: Int) {
                (context as Activity).startActivityForResult(Intent(context, LoginActivity::class.java), reqCode)
            }

            override fun onGetUserInfo(): UserInfo? {
                if (phone == null || token == null) {
                    return null
                }
                var userInfo = UserInfo(token, "", phone, "")
                return userInfo
            }

            override fun onShare(context: Context?, shareInfo: ShareInfo?) {
                toast("调起宿主分享")
            }

            override fun onPay(context: Context?, jsInterface: CommunityJavaScriptInterface?, responseId: String?, payment: CommunityPayment?) {
                sJsInterface = jsInterface
                sResponseId = responseId
                //跳转订单支付页面
                var intent: Intent? = Intent(context, PayActivity::class.java)
                intent?.putExtra("responseId", responseId)
                intent?.putExtra("orderId", payment?.param)
                intent?.putExtra("payMoney", payment?.amount)
                context?.startActivity(intent)
                //支付结束后回调（成功 失败 取消）
//                MyApplication.sJsInterface?.callPayed(MyApplication.sResponseId, CommunityConstants.SDK_SUCCESS)
            }
        })

    }

}